### PR TITLE
Add data limit according to platform

### PR DESCRIPTION
**Pull request by devkonrad**
Ads data constraint according to platform, for URLs and other fields 

Changes:
modified: app/models/team_submission.rb

Refs: https://github.com/Iridescent-CM/technovation-app/issues/3372

### CODE REVIEW

**Comment by devkonrad** [comment]
As discussed with @dboyer the "App Inventor Name" need to raise an error when its filled and we select "thunkable" as platform.

**Comment by devkonrad** [comment]
> Ads data constraint according to platform, for URLs and other fields
> 
> Changes: modified: app/models/team_submission.rb
> 
> Refs: #3372

**QA Test Steps:**
_Enter to edit team submission page (on admin)_
1 - When develop platform changed to thunkable:
	- If you try to fill de App Inventor Name you will get an error
	- If you try to save without Thunkable URL you will get an error
2 - When develop platform changed to thunkable:
	- If you try to save with a Thunkable URL you will get an error
3 - If you try to save invalid URLs value on Demo video Link you will get an error
4 - If you try to save invalid URLs value on Pitch video Link you will get an error

File: app/models/team_submission.rb [hunk 1]
```diff
@@ -593,6 +612,20 @@ def question_response(question)
 
   private
 
+  def padronize_url( url )
```
Comment (viviancan) [inline app/models/team_submission.rb:615]: Hey Alex! Thanks for all the work on this one! 

It looks like this method name is in Portuguese (padronize). Could you update it to the English equivalent maybe something along the lines of `standardize_url` or something similar, in order to keep the consistency with English naming throughout the application?

**Review by viviancan — CHANGES_REQUESTED** [review]
Hi Alex, I left a few followup notes on the issue #3372. I had a few issues with editing a submission.

**Comment by devkonrad** [comment]
> Hi Alex, I left a few followup notes on the issue #3372. I had a few issues with editing a submission.

Hi, Vivian! I answer your question about the validation here https://github.com/Iridescent-CM/technovation-app/issues/3372 . I just remain the validation that was use, but we can impove that if needed. :)

File: app/validators/thunkable_share_url_validator.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class ThunkableShareUrlValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
-    if !value.match(/^https?:\/\/x.thunkable.com\/projects\/.+$/)
```
Comment (viviancan) [inline app/validators/thunkable_share_url_validator.rb:3]: The "project" version is actually correct here! Sry about the confusion Alex!

**Review by viviancan — CHANGES_REQUESTED** [review]
Hi Alex, thanks for tackling this 👍 After verifying w/ deb this morning, I realized I misunderstood the requirement for the the thinkable url this morning. The original version with "project" is actually the correct version of the link we are accepting now - not the version with "copy". The change will have to be reverted back to the original URL that includes "project" - I think  in file `app/validators/thunkable_share_url_validator.rb` and the spec file.

**Review by viviancan — CHANGES_REQUESTED** [review]
Hey Alex, LGTM! Tested it locally and it is working as expected.

The only thing is on `def standardize_url( url )` there is extra whitespace around `url`.

**Comment by devkonrad** [comment]
> Hey Alex, LGTM! Tested it locally and it is working as expected.
> 
> The only thing is on `def standardize_url( url )` there is extra whitespace around `url`.

Great, @viviancan! Thanks for the review. I will fix it and merge to QA. :)

**Comment by dboyer** [comment]
@devkonrad Feel free to merge this to QA if it's just reverting the previous change.

**Comment by devkonrad** [comment]
> @devkonrad Feel free to merge this to QA if it's just reverting the previous change.

Great. Doing that. :)

**Pull request merged**